### PR TITLE
Added module for highlight users you follow and streamer in the chat

### PR DESCRIPTION
**Pull request by im1x**

### CODE REVIEW

File: src/modules/highlight_follow_streamer_chat/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+const twitchAPI = require('../../utils/twitch-api');
+
+const STYLE_HIGHLIGHT_FOLLOW = 'bttv-highlight-follow-nick';
+const STYLE_HIGHLIGHT_STREAMER = 'bttv-highlight-streamer-nick';
+
+const followsArray = [];
+let currentChannelName;
+let secondRequestAllowed = true;
+
+function putFollowsInArray(direction, limit) {
```
Comment (night) [inline src/modules/highlight_follow_streamer_chat/index.js]: i would not feel comfortable with the amount of requests this generates.. every page load could cause many records being loaded from twitch, and this doesn't even cache that data

File: src/modules/highlight_follow_streamer_chat/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        settings.on('changed.highlightFollowAndStreamerChat', () => this.load());
+        watcher.on('chat.message', ($message, messageObj) => this.onMessage($message, messageObj));
+        watcher.on('vod.message', $message => this.onVodMessage($message));
+        watcher.on('load.chat', () => this.load());
+        watcher.on('load.vod', () => this.load());
+    }
+
+    load() {
+        if (settings.get('highlightFollowAndStreamerChat') === false) return;
+        currentChannelName = twitch.getCurrentChannel().name;
```
Comment (night) [inline src/modules/highlight_follow_streamer_chat/index.js]: we should make this feature just followed channels, not streamer too .. we can probably assume that if you care about seeing the streamer in chat you'd follow their channel

Comment (im1x) [reply src/modules/highlight_follow_streamer_chat/index.js]: @night Then we can't highlight streamer's nickname with certain CSS. But if you want, I'll do it.

File: src/modules/highlight_follow_streamer_chat/index.js (hunk 1 above)
Comment (im1x) [reply src/modules/highlight_follow_streamer_chat/index.js]: @night I can add a cache. But will this be enough? There will still be many requests.
You need to decide. If the cache helps with this problem, then I'll do it. But if it doesn’t make sense, then I can just close the pull request.

**Comment by night** [comment]
gonna close this since I don't really want that many requests being generated. if twitch ever exposes this data in chat it would be easily doable, but otherwise not the greatest feature for now.

**Pull request closed without merging**